src/java/main/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -71,7 +71,7 @@
     NettyServerCnxnFactory factory;
     boolean initialized;
     
-    NettyServerCnxn(Channel channel, ZooKeeperServer zks, NettyServerCnxnFactory factory) {
+    public NettyServerCnxn(Channel channel, ZooKeeperServer zks, NettyServerCnxnFactory factory) {
```

Comment:
@yufeldman Thanks for explaining, makes sense to me. It is required to run the tests, however I do find another problem with tests: it looks like the Netty tests (testNetty*) never run with current configuration. Proof: remove the public access specifier appertain to `NettyServerCnxn` and all tests of `ServerCxnExceptionsTest` still pass. We expect Netty related tests fail here without public access specifier, right? Now put back the public for `NettyServerCnxn` but remove the public access specifier appertains to `NIOServercCnxn`, now all tests failed while we expect only NIO tests fail but Netty tests pass. 

It's likely caused by the intervening of the java system properties that controls the Netty vs NIO server selection. One solution is to split the `ServerCxnExceptionsTest` into Netty and NIO specific tests.